receiver/hostmetricsreceiver/hostmetrics_receiver.go

```diff
@@ -48,42 +54,101 @@ func NewHostMetricsReceiver(
 			return nil, fmt.Errorf("host metrics scraper factory not found for key: %s", key)
 		}
 
-		scraper, err := factory.CreateMetricsScraper(ctx, logger, cfg, consumer)
+		scraper, err := factory.CreateMetricsScraper(ctx, logger, cfg)
 		if err != nil {
 			return nil, fmt.Errorf("cannot create scraper: %s", err.Error())
 		}
 		scrapers = append(scrapers, scraper)
 	}
 
-	hmr := &Receiver{
+	hmr := &receiver{
 		config:   config,
 		scrapers: scrapers,
+		consumer: consumer,
 	}
 
 	return hmr, nil
 }
 
-// Start begins scraping host metrics based on the OS platform.
-func (hmr *Receiver) Start(ctx context.Context, host component.Host) error {
+// Start initializes the underlying scrapers and begins scraping
+// host metrics based on the OS platform.
+func (hmr *receiver) Start(ctx context.Context, host component.Host) error {
+	ctx, hmr.cancel = context.WithCancel(ctx)
+
+	go func() {
+		hmr.initializeScrapers(ctx, host)
+		hmr.startScrapers(ctx)
+	}()
+
+	return nil
+}
+
+// Shutdown terminates all tickers and stops the underlying scrapers.
+func (hmr *receiver) Shutdown(ctx context.Context) error {
+	hmr.cancel()
+	return hmr.closeScrapers(ctx)
+}
+
+func (hmr *receiver) initializeScrapers(ctx context.Context, host component.Host) {
+	for _, scraper := range hmr.scrapers {
+		err := scraper.Initialize(ctx)
+		if err != nil {
+			host.ReportFatalError(err)
+			return
+		}
+	}
+}
+
+func (hmr *receiver) startScrapers(ctx context.Context) {
 	go func() {
-		for _, scraper := range hmr.scrapers {
-			err := scraper.Start(ctx)
-			if err != nil {
-				host.ReportFatalError(err)
+		ticker := time.NewTicker(hmr.config.CollectionInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				hmr.ScrapeMetrics(context.Background(), hmr.scrapers)
+			case <-ctx.Done():
 				return
 			}
 		}
 	}()
+}
 
-	return nil
+func (hmr *receiver) ScrapeMetrics(ctx context.Context, scrapers []internal.Scraper) {
+	ctx, span := trace.StartSpan(ctx, "hostmetricsreceiver.ScrapeMetrics")
+	defer span.End()
+
+	metricData := data.NewMetricData()
+	metrics := internal.InitializeMetricSlice(metricData)
+
+	var errors []error
+	for _, scraper := range scrapers {
+		scraperMetrics, err := scraper.ScrapeMetrics(ctx)
+		if err != nil {
+			errors = append(errors, err)
```

Comment:
Continue here? Please add a test :)

Reply:
There's intentionally no `continue` as a Scraper may fail to scrape some metrics, but succeed to scrape others. This is documented [here](https://github.com/open-telemetry/opentelemetry-collector/pull/952/files#diff-6b42a9a8648fce4026a8c7229e4d9267R37). As far as I can tell this is the most idiomatic way to handle this in Go but let me know if there's a better way.

I do still need to add several more tests around failure cases though. Preferably in a followup PR :)